core/transport.js

```diff
@@ -92,12 +94,16 @@ module.exports = function (driver) {
 
   function setMu (muInstance) {
     mu = muInstance
+    logger = mu.log.child({ muid })
```

Comment:
the problem is how mu is passed around. Not the logger itself, it's another PR to address that issue.